### PR TITLE
Allow dumping debug info via sys topic.

### PR DESCRIPTION
**Pull request by aforge**
Please take a look and lmk what you think.
Things like this will be helpful for diagnosing various problems with the server.

### CODE REVIEW

**Review by or-else — COMMENTED** [review]
Why not dump it into `expvars`? It would seem to be more logical because it already exposes some of the internal state.

**Comment by aforge** [comment]
> Why not dump it into `expvar`? It would seem to be more logical because it already exposes some of the internal state.

`expvar` is accessed via http which allows little control over who can access this data.
Plus, `expvar` is supposed to be used for exporting various internal stats, not the internal state. The latter is more suitable for some kind of a status page.

**Comment by or-else** [comment]
> expvar is accessed via http which allows little control over who can access this data.

I'm not too concerned. It can be disabled entirely, it can be easily controlled at the nginx level. 

My main concern is that using `sys` for sending debug data opens a new venue into the server. For instance, think of a production setup. This new approach creates a requirement to give root access to those who debug the server. Or things which are not controlled by normal access permissions. Maybe we should implement the whole z: statz, healthz, varz, etc.

Do you want to talk by voice?

**Comment by aforge** [comment]
I've added a '/status' endpoint for exposing debug data per discussion.
This will require some more work. But the core functionality is this.

**Review by or-else — APPROVED** [review]

File: server/http.go [hunk 1]
```diff
@@ -366,3 +368,61 @@ func authHttpRequest(req *http.Request) (types.Uid, []byte, error) {
...
 }
+
+func serveStatus(wrt http.ResponseWriter, req *http.Request) {
+	// Disable caching.
+	wrt.Header().Set("Cache-Control", "no-cache, no-store, private, max-age=0")
+	wrt.Header().Set("Expires", time.Unix(0, 0).Format(http.TimeFormat))
+	wrt.Header().Set("Pragma", "no-cache")
+	wrt.Header().Set("X-Accel-Expires", "0")
+
+	fmt.Fprintf(wrt, "<html><head>")
```
[the remark this answers is not in the document]
Comment (aforge) [reply server/http.go:379]: We need human readable data anyway. So html or json, it makes no difference to us.

File: server/main.go [hunk 2]
```diff
@@ -616,6 +616,7 @@ func main() {
 		}
 	}
 	logs.Info.Printf("API served from root URL path '%s'", config.ApiPath)
+	mux.HandleFunc("/status", serveStatus)
```
[the remark this answers is not in the document]
Comment (aforge) [reply server/main.go]: Done.

**Pull request merged**